### PR TITLE
Add topics to repos with api failures

### PR DESCRIPTION
**Pull request by snim2**
This PR changes the way we deal with failures from the wordpress.org API. Previously, we raised an exception and carried on. Now, we "annotate" the plugin by adding repository topics that explain why the plugin is not available on wordpress.org and the `skip-mirror` topic, so that we don't try to update from WordPress again.

Repo topics are of the form:

* `status:<string>` for error messages, e.g. `status:closed` or `status:plugin-not-found`
* `reason:<string>` for the `reason` key in JSON, e.g. `reason:security-issue`, `reason:merged-into-core`, etc.

The intention here is that prefixes are unique, i.e. each repository should have only one `status` or `reason` topic set, and any old topics of the same "type" are removed before adding a new one.

Later on, this convention will help us add topics like `installs:N` to say how many times we have installed each plugin.

Setting repository topics also allows us to run a search for all plugins that were closed for a particular reason.

We also add two new environment variables for convenience:

* `ANNOTATE_SKIPPED_REPOS` allows repos that already have the `skip-mirror` topic set to be tagged. The current repos that have this were tagged by hand, and so will only get `status` and `reason` topics with this env var set.
* `FORCE_ANNOTATION` ensures that the script adds and removes repository topics even in `dry-run` mode, so that tagging can be de-coupled from pushing to repositories.

The intention is for these env vars to be set when the script is run manually, so they are always `false` in the GitHub actions and example `.env` file.

This PR also contains some refactoring commits, so it might be easier to review commit-by-commit.

## Testing

Make sure you have a .env that matches the _new_ `.env.wordpress-example` and that `DRY_RUN` is set to true, and the new env vars are set to false, then run:

```shell
./mirror-wordpress-plugins --dry-run
```

You might also want to comment out the lines in the `Plugin.cleanup` method so that you can see the tags and commits that the script created.

### CODE REVIEW

File: mirror-wordpress-plugins [hunk 1]
```diff
@@ -145,8 +177,13 @@ class Plugin
     rescue JSON::ParserError => e
       raise WordPressPluginApiError, e.message
     end
-    # Whippet Racetrack error message here was "No plugin info available for <plugin>".
-    raise WordPressPluginApiError, plugin_info["error"].to_s if plugin_info.key?("error")
+    if plugin_info.key?("error")
```
Comment (RobjS) [inline mirror-wordpress-plugins:180]: If the Api response does not include an error, do we want to remove the `skip-mirror` topic (and any other errors)? I'm assuming it's hypothetically possible that a plugin could be closed and subsequently re-opened, but I don't know for sure if that's the case.

**Review by RobjS — CHANGES_REQUESTED** [review]
All seems to be working as expected, just a few suggestions for minor tweaks/refactors.

Not one for now, but I wonder if this is getting to the point where we should split it out across multiple files? I'm finding the single-file format quite hard to follow as it grows more complex.

File: mirror-wordpress-plugins (hunk 1 above)
Comment (snim2) [reply mirror-wordpress-plugins:180]: I'm in two minds about this, because the automated script won't be able to tell whether `skip-mirror` was added for that reason or whether a develop added the topic manually. Maybe it would be better to add an `active-on-wordpress` topic, then we can search for repos that have both?

File: mirror-wordpress-plugins [hunk 2]
```diff
@@ -89,11 +89,11 @@ class PluginUpdater
     if !prefix.empty?
       @topics.each do |topic|
         if topic.start_with?("#{prefix}:")
-          RepositoryTopic.new(topic.split(":")[1], prefix).remove_from_repo(@repo)
+          RepositoryTopic.new(topic.split(":")[1], prefix).remove_from_repo()
```
Comment (RobjS) [inline mirror-wordpress-plugins]: Should we be passing in the repo reference here?

Comment (snim2) [reply mirror-wordpress-plugins]: This ended up in [c16ad26](https://github.com/dxw-wordpress-plugins/mirror-gitlab-plugins/pull/24/commits/c16ad26841b32ef19834836c51a6aa83564c3741) and will need squashing

**Review by RobjS — APPROVED** [review]

**Comment by snim2** [comment]
Squashing the linter fix commit before merge
